pkg/utils/cache/ttl.go

```diff
@@ -0,0 +1,68 @@
+package cache
+
+import (
+	"time"
+
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/trisacrypto/directory/pkg/bff/config"
+)
+
+// TTL is a wrapper around a thread-safe LRU cache that adds a time-to-live value to
+// each entry.
+type TTL struct {
+	conf config.CacheConfig
+	lru  *lru.Cache
+}
+
+func NewTTL(conf config.CacheConfig) (cache *TTL, err error) {
+	cache = &TTL{
+		conf: conf,
+	}
+
+	if cache.lru, err = lru.New(int(conf.Size)); err != nil {
+		return nil, err
+	}
+
+	return cache, nil
+}
```

Comment:
Let's add the enabled/disabled behavior down to the `cache` module rather than have the external caller decide.

```suggestion
func New(conf config.CacheConfig) (Cache, error) {
	if !conf.Enabled {
		return &Disabled{}, nil
	}
	return NewTTL(conf)
}

func NewTTL(conf config.CacheConfig) (cache *TTL, err error) {
	cache = &TTL{
		conf: conf,
	}

	if cache.lru, err = lru.New(int(conf.Size)); err != nil {
		return nil, err
	}

	return cache, nil
}
```